From aede3514413a62d1173ac59962d9fc2620021e31 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Sat, 21 Nov 2015 21:22:50 +0800
Subject: [PATCH] rk: arm64: support build kernel.img/resource.img/[z]boot.img
 with ramdisk

zboot.img: Image.lz4 + resouorce.img
boot.img:  Image + resource.img

The ramdisk name should be ramdisk.img.
The ramdisk.img should be placed on $(objtree).
Support symbolic link.

Change-Id: Iaf50be16e321b13bbe45b9beb335d542e00507cf
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm64/Makefile | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 35649ee8ad56..e188c2ce6659 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -168,3 +168,33 @@ define archhelp
   echo  '                  (distribution) /sbin/installkernel or'
   echo  '                  install to $$(INSTALL_PATH) and run lilo'
 endef
+
+kernel.img: Image.lz4
+	$(Q)scripts/mkkrnlimg $(objtree)/arch/arm64/boot/Image $(objtree)/kernel.img >/dev/null
+	@echo '  Image:  kernel.img is ready'
+
+LOGO := $(notdir $(wildcard $(srctree)/logo.bmp))
+LOGO_KERNEL := $(notdir $(wildcard $(srctree)/logo_kernel.bmp))
+RAMDISK_IMG := $(notdir $(wildcard $(objtree)/ramdisk.img))
+make_boot_img := \
+	$(srctree)/scripts/mkbootimg \
+		--kernel $(objtree)/arch/arm64/boot/Image \
+		$(if $(RAMDISK_IMG),--ramdisk $(objtree)/ramdisk.img) \
+		--second resource.img \
+		-o boot.img && \
+	echo '  Image:  boot.img (with Image $(if $(RAMDISK_IMG),'ramdisk.img ')resource.img) is ready'; \
+	$(srctree)/scripts/mkbootimg \
+		--kernel $(objtree)/arch/arm64/boot/Image.lz4 \
+		$(if $(RAMDISK_IMG),--ramdisk $(objtree)/ramdisk.img) \
+		--second resource.img \
+		-o zboot.img && \
+	echo '  Image:  zboot.img (with Image.lz4 $(if $(RAMDISK_IMG),'ramdisk.img ')resource.img) is ready'
+
+%.img: rockchip/%.dtb kernel.img $(LOGO) $(LOGO_KERNEL)
+	$(Q) if [ -f $(srctree)/$(LOGO) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO) $(objtree)/; fi
+	$(Q) if [ -f $(srctree)/$(LOGO_KERNEL) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO_KERNEL) $(objtree)/; fi
+	$(Q)scripts/resource_tool $(objtree)/arch/arm64/boot/dts/rockchip/$*.dtb $(LOGO) $(LOGO_KERNEL)
+	@echo '  Image:  resource.img (with $*.dtb $(LOGO) $(LOGO_KERNEL)) is ready'
+	$(Q) if [ -x $(srctree)/scripts/mkbootimg ]; then $(make_boot_img); fi
+
+CLEAN_FILES += boot.img kernel.img resource.img zboot.img
-- 
2.35.3

